File: D6(dynamic)/fibo.py
def fib_memo(n):
    arr = [None] * (n+1)
    arr[1] = 1
    arr[2] = 1

    i = 3
    while i <= n:
        arr[i] = arr[i-1] + arr[i-2]
        i += 1
    print(arr[n])

fib_memo(3)
fib_memo(5000000)

File: D6(dynamic)/fibo.c
#include <stdio.h>
#define MAX 10000
int arr[MAX];

int main() {
    arr[1] = 1;
    arr[2] = 1;
    for(int i=3; i < MAX; i++) {
        arr[i] = arr[i-1] + arr[i-2];
    }
    //대충 이런식
    return 0;
}